src/main/java/com/google/devtools/build/lib/collect/nestedset/Depset.java

```diff
@@ -97,6 +101,11 @@
             + " may interfere with the ordering semantics.")
 @Immutable
 public final class Depset implements StarlarkValue, Debug.ValueWithDebugAttributes {
+  private static final ImmutableMap<Order, Depset> EMPTY_DEPSETS =
+      Arrays.stream(Order.values())
+          .map(Order::emptySet)
+          .collect(toImmutableEnumMap(NestedSet::getOrder, e -> new Depset(null, e)));
+
```

Comment:
Is it possible to follow the same pattern as for NestedSets? That is to add `emptyDepset` next to `emptySet` to `Order` class.

That would satisfy the requirement to have an empty depset interned for each Order element. 

It also seems not to create a problem with a cycle, because Depset, Nestedset and Order are in the same `java_library`.

Implementing it in Order, would make the code a bit simpler (not introducing 2 different implementation for NestedSet and Depset) and making it easier to follow.

Reply:
Done. This required some further changes to break cyclic dependencies between `Depset`'s and `Order`'s static initializers. 

Let me know if I should also add `@SerializationConstant`s for the empty depsets - I don't know whether they could be subject to serialization.